src/util/fs_helpers.cpp

```diff
@@ -298,3 +303,15 @@ std::optional<fs::perms> InterpretPermString(const std::string& s)
         return std::nullopt;
     }
 }
+
+#ifdef __APPLE__
+FSType GetFilesystemType(const fs::path& path)
```

Comment:
nit: this may still be a bit too general for the current problem, I'm fine with it as it is, but if you touch again consider:
```C++
bool IsExfatFilesystem(const std::filesystem::path& p)
```
Which would simplify the PR even more.